sourcegraph/sourcegraph
# Code Insights: Unify lang stats insight fetching logic

Closes https://github.com/sourcegraph/sourcegraph/issues/31085

## Context

Prior to this PR we had two different gql handlers for fetching data for the lang stats insight (one for live preview on the creation UI, one on the dashboard page). In this PR we extract logic about lang stats insight from legacy CodeInsightsBackend class API to data fetching in one fetching hook (see this tracking issue to get more context on legacy class based API) https://github.com/sourcegraph/sourcegraph/issues/41145

## Test plan
- Check that lang stats card works as expected on the dashboard page
- Check lang stats creation UI live preview chart
- Check lang stats insight standalone insight page (to do it click on the lang stats insight card title on the dashboard page)



## App preview:

- [Web](https://sg-web-vk-lang-stats-insight-preview.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
